src/app/layout/header/menus.service.ts

```diff
@@ -12,46 +12,142 @@ export class MenusService {
   readonly menus: Map<ContextType, MenuItem[]>;
 
   constructor() {
-    this.menus = new Map<ContextType, MenuItem[]>([
-      [
-        ContextTypes.BUILTIN.get('space'),
-        [
+
+    this.menus = new Map<ContextType, MenuItem[]>();
+    let menuItems: MenuItem[] = [
```

Comment:
Instead of initializing an array here it would be nice to have a way of saying that a given menu item is supposed to be sorted under another menu item. That way it is less hard to mess with the data structure.

Reply:
I agree. Furthermore, I'd vote for something that is actually registering available components and assembles the structure. This hardcoding is bad.